### ISSUE TITLE
401 error received at page refresh for users who deleted their accounts on FxA

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-frontend#9482]
Follow up for https://github.com/mozilla/addons-server/issues/14702#issuecomment-648846438
Prerequisites: `fxa-account-delete` waffle is enabled

### Describe the problem and steps to reproduce it:
1. Log in on AMO stage (the FxA server is more reliable here)
2. Go to the Edit profile page and click on Manage Firefox account (open this link in a new tab)
3. Delete the user account from FxA
4. Go back to AMO and refresh the page

### What happened?
There is a 401 error thrown when the page is refreshed. This happens for every page on the frontend for as long as the browser session is open.

### What did you expect to happen?
There should be no error received when the page is refreshed 

![2020-06-24_17-25-50](https://user-images.githubusercontent.com/31961530/85577156-ad9b7f80-b641-11ea-9de9-81587c1c7c79.gif)

**Comment by willdurand** [comment]
This is because we're not clearing the cookies and the auth cookie is sent to the server on page refresh.

**Comment by willdurand** [comment]
On a second thought, I think we might be able to solve the issue in addons-frontend. I am not sure how we will display a message to the user yet but we should be able to:

1. catch the 401 when loading the user profile on the server
2. force-delete the auth cookie
3. notify user

For (2), hopefully it is doable _via_ our express server.

For (3), not sure we can reuse our UI error handling. We might have to introduce a special flag in a redux sate to deal with this error. Maybe we could add something in `errorPage`.

**Comment by AlexandraMoga** [comment]
@willdurand I've verified the fix on stage and I can confirm that I no longer receive 401s when refreshing any frontend pages. There is also a message displayed to users to inform them they've been logged out.

There are some edge cases caused by the frontend-auth cookie being deleted. I've filed #9490 and #9491 as follow-ups since it's easier to give STR that way. 

[homepage]
![image](https://user-images.githubusercontent.com/31961530/86231270-f780ea00-bb9a-11ea-8a57-a0278dc8b70b.png)

[user profile edit page - page that requires authentication]
![image](https://user-images.githubusercontent.com/31961530/86231293-01a2e880-bb9b-11ea-895e-69733ae38dde.png)
